daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Error.scala

```diff
@@ -44,6 +45,14 @@ final case class ContractNotFound(ci: ContractId) extends Error {
   override def detailMsg: String = msg
 }
 
+/** See com.daml.lf.transaction.Transaction.DuplicateContractKey
+  *    for more information.
+  */
+final case class DuplicateContractKey(key: GlobalKey) extends Error {
```

Comment:
@moritzkiefer-da : do you think it would be possible / sensible to also report the contract-ids on which the collision occurs?

Reply:
> @moritzkiefer-da : do you think it would be possible / sensible to also report the contract-ids on which the collision occurs?

Maybe but we don’t do that atm in the ledgers that do this and for now I’m interested in getting consistency and fixing the bug not in exposing more information.

Reply:
Ack. Agreed.

Reply:
Sorry for raising this unnecessarily.